commit ab6e9f96a5147a49e19d5ce216761202779a88d6
Author: Marcello Sachs <marcello.sachs@gmail.com>
Date:   Mon Jun 29 16:18:43 2015 -0400

    load assets over ssl

diff --git a/app/assets/stylesheets/custom.css.scss b/app/assets/stylesheets/custom.css.scss
index 8679a31..7caeea8 100644
--- a/app/assets/stylesheets/custom.css.scss
+++ b/app/assets/stylesheets/custom.css.scss
@@ -1,4 +1,4 @@
-@import url(http://fonts.googleapis.com/css?family=Open+Sans:700,400,300,600);
+@import url(https://fonts.googleapis.com/css?family=Open+Sans:700,400,300,600);
 
 .flash-notice {
   color: red;
diff --git a/app/views/application/_head.html.erb b/app/views/application/_head.html.erb
index 55677b8..dd6398d 100644
--- a/app/views/application/_head.html.erb
+++ b/app/views/application/_head.html.erb
@@ -6,7 +6,7 @@
 <%= render partial: 'typekit' %>
 <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track' => true %>
 <%= csrf_meta_tags %>
-<link href="http://fonts.googleapis.com/css?family=Comfortaa:400,700" rel="stylesheet" type="text/css" />
+<link href="https://fonts.googleapis.com/css?family=Comfortaa:400,700" rel="stylesheet" type="text/css" />
 
 <%= render partial: 'application/head_embed_codes' %>
 
diff --git a/app/views/layouts/application.html.erb b/app/views/layouts/application.html.erb
index 32a29f2..90be66e 100644
--- a/app/views/layouts/application.html.erb
+++ b/app/views/layouts/application.html.erb
@@ -13,7 +13,7 @@
       <!-- Got sick of flaky Google CDN for dev -->
       <%= javascript_include_tag 'jquery-1.8.2' %>
     <% else %>
-      <script src="http://ajax.googleapis.com/ajax/libs/jquery/1.8.2/jquery.min.js"></script>
+      <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.8.2/jquery.min.js"></script>
     <% end %>
     <!-- Include all compiled plugins (below), or include individual files as needed -->
     <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
diff --git a/app/views/layouts/home.html.erb b/app/views/layouts/home.html.erb
index 849b222..a8c85a5 100644
--- a/app/views/layouts/home.html.erb
+++ b/app/views/layouts/home.html.erb
@@ -14,7 +14,7 @@
   <%= render partial: 'footer' %>
 
     <!-- jQuery (necessary for Bootstrap's JavaScript plugins) -->
-    <script src="http://ajax.googleapis.com/ajax/libs/jquery/1.8.2/jquery.min.js"></script>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/1.8.2/jquery.min.js"></script>
     <!-- Include all compiled plugins (below), or include individual files as needed -->
     <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
 </body>
